### ISSUE TITLE
Python 3.x: beware of floating division

### ISSUE THREAD

**Issue by stefandrissen** [simonowen/pyz80#11]
Python 3.x performs floating division by default, whereas Python 2.x performed integer division by default.

The work-around is to use // in source code instead of /
````
autoexec
dump 1,0
org 32768

ld a,127 // 256	; -> 0
ld a,128 // 256	; -> 0

ld a,127 / 256	; -> 0
ld a,128 / 256	; -> 1

ret
````

Since floating division seems somewhat weird in Z80 code, it may be sensible to replace / with // while assembling (see def parse_expression) with something like:

    arg = arg.replace('/','//')

**Comment by simonowen** [comment]
Good find!  That'd cause some pretty subtle bugs...

The only potential catch I can think of is that the // operator behaves differently with negative numbers between Python 2 and 3 (see [this](https://python-history.blogspot.com/2010/08/why-pythons-integer-division-floors.html)/  That's probably not going to be an issue for assembled code though.

It may be worth ensuring only single slashes are replaced too?

`
arg = re.sub(r"\b/\b", "//", arg)
`

**Comment by stefandrissen** [comment]
Fortunately my crashes were far from subtle :-)

Another option could be to only do this when in "comet" mode and add an item to the help page highlighting this behavior.

**Comment by simonowen** [comment]
Looking at the code, the rounding away from zero seems to be intentional:

```python
    farg = eval(argcopy)
    if farg >= -.5:
        farg += 0.5
    else:
        farg -= 0.5
    narg = int(farg)
```

It seems like replacing all that with   `narg = int(eval(argcopy))`  _should_ give consistent integer truncation for Python 2.x and 3.x.

Rounding by default probably breaks more than it helps in old code, where Comet didn't support floating point.  Anywhere that still wants float rounding can just add 0.5, and/or use floor() to round up away from zero.

**Comment by simonowen** [comment]
Finally fixed in 1a1a639bdb3d1753b61449c5cf53dd73c337d40e. The test script needed the floor() update, which was fixed in 98d14c1c60300536c06e165daaeb6c8e1c0cc877.
